src/graph_gui.cpp

```diff
@@ -493,12 +505,18 @@ struct BaseGraphWindow : Window {
 			byte month = this->month;
 			Year year  = this->year;
 			for (int i = 0; i < this->num_on_x_axis; i++) {
-				SetDParam(0, month + STR_MONTH_ABBREV_JAN);
-				SetDParam(1, month + STR_MONTH_ABBREV_JAN + 2);
-				SetDParam(2, year);
-				x_label_width = max(x_label_width, GetStringBoundingBox(month == 0 ? STR_GRAPH_X_LABEL_MONTH_YEAR : STR_GRAPH_X_LABEL_MONTH).width);
+				if (this->month_increment > 1) {
```

Comment:
is the special case needed here? It's so nearly identical...

Reply:
It looks a bit weird when month_increment is 1...